src/App.tsx

```diff
@@ -1,27 +1,86 @@
-import React from 'react';
+import React, { useState } from 'react';
+import 'bulma/css/bulma.min.css';
 import './App.scss';
+
+import { Good } from './types/Good';
+import { getAll, get5First, getRedGoods } from './api/goods';
+
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [isError, setIsError] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const throwError = () => {
+    setIsError(true);
+    setIsLoading(false);
+  };
+
+  const getAllGoods = () => {
+    setIsLoading(true);
+    getAll()
+      .then(result => {
+        setGoods(result);
+        setIsLoading(false);
+      })
+      .catch(throwError);
+  };
+
+  const get5Goods = () => {
+    get5First()
+      .then(result => setGoods(result))
+      .catch(throwError);
+  };
+
+  const getOnlyRedGoods = () => {
+    getRedGoods()
+      .then(result => setGoods(result))
+      .catch(throwError);
+  };
+
+  if (isError) {
+    return (
+      <p className="notification is-danger">
+        Oops! Something went wrong!
+      </p>
+    );
+  }
```

Comment:
This part breaks your app. Display an error only instead of the GoodsList component. The user needs to have the possibility to click the buttons again to retry.